torch/distributed/distributed_c10d.py

```diff
@@ -301,41 +301,67 @@ def _warn_not_in_group(op_name):
     )
 
 
-def _get_group_rank(group: ProcessGroup, rank):
+def get_local_rank(group: ProcessGroup, global_rank: int) -> int:
     """
-    Helper that gets a given group's local rank in the group from a given global
-    rank.
+    Translate a global rank into a local rank.
+
+    ``global_rank`` must be part of ``group`` otherwise this raises RuntimeError.
+
+    Args:
+        group (ProcessGroup): ProcessGroup to find the relative rank.
+        global_rank (int): Global rank to query.
+
+    Returns:
+        Local rank of ``global_rank`` relative to ``group``
+
+    N.B. calling this function on the default process group returns identity
     """
     if group is GroupMember.WORLD:
-        raise RuntimeError(
-            "group.WORLD does not have local rank to global " "rank mapping"
-        )
+        return global_rank
     if group not in _pg_group_ranks:
-        raise RuntimeError("The given group does not exist")
-    try:
-        group_rank = _pg_group_ranks[group][rank]
-    except KeyError:
-        raise RuntimeError(
-            f"The global rank {rank} is not part of the group {group}"
-        ) from None
-    return group_rank
+        raise RuntimeError("Group {group} is not registered with torch.distributed")
+    group_ranks = _pg_group_ranks[group]
+    if global_rank not in group_ranks:
+        raise RuntimeError(f"Global rank {global_rank} is not part of group {group}")
 
+    return group_ranks[global_rank]
 
-def _get_global_rank(group, group_rank):
+def get_global_rank(group: ProcessGroup, local_rank: int) -> int:
     """
-    Helper that gets a given group's global rank from a given local rank in the
-    group.
+    Translate a local rank into a global rank.
+
+    ``local_rank`` must be part of `group` otherwise this raises RuntimeError.
+
+    Args:
+        group (ProcessGroup): ProcessGroup to find the global rank from.
+        local_rank (int): Local rank to query.
+
+    Returns:
+        Global rank of ``local_rank`` relative to ``group``
+
+    N.B. calling this function on the default process group returns identity
     """
     if group is GroupMember.WORLD:
-        raise RuntimeError(
-            "group.WORLD does not have local rank to global " "rank mapping"
-        )
-    group_rank_map = _pg_group_ranks[group]
-    for rank, grp_rank in group_rank_map.items():
-        if grp_rank == group_rank:
+        return local_rank
+    if group not in _pg_group_ranks:
+        raise RuntimeError("Group {group} is not registered with torch.distributed")
+    for rank, grp_rank in _pg_group_ranks[group].items():
+        if grp_rank == local_rank:
             return rank
-    raise RuntimeError("The group rank is not part of the group")
+    raise RuntimeError(f"Global rank {local_rank} is not part of group {group}")
 
+def get_global_ranks(group: ProcessGroup) -> Sequence[int]:
```

Comment:
It can be used to create a new process group based on an existing one.
For example:
```
dist.new_group(ranks=dist.get_group_ranks(nccl_pg), backend="gloo")
```

Right now this PR doesn't provide all accessors needed for this API to be general purpose enough, one can't access timeouts and options yet.